support-frontend/assets/helpers/internationalisation/country.ts

```diff
@@ -627,16 +627,17 @@ function detect(
 	];
 	let country: IsoCountry | null = null;
 
-	targetCountryGroups.forEach((targetCountryGroupId) => {
+	for (const targetCountryGroupId of targetCountryGroups) {
 		const candidateCountry = handleCountryForCountryGroup(
 			targetCountryGroupId,
 			countryGroupId,
 		);
 
-		if (candidateCountry !== null && country === null) {
+		if (candidateCountry !== null) {
 			country = candidateCountry;
+			break;
 		}
-	});
+	}
```

Comment:
With the original version (using the `forEach` method) eslint was complaining about the conditional on line 642. By changing to a `for...of` loop eslint can now tell `country` is getting assigned before the check. I also think this is slightly better as we can now `break` from the loop early instead of having a conditional check